packages/plugins/users-permissions/admin/src/pages/Roles/ListPage/tests/index.test.js

```diff
@@ -13,6 +13,8 @@ jest.mock('@strapi/helper-plugin', () => ({
   useTracking: jest.fn(() => ({ trackUsage: jest.fn() })),
   useNotification: jest.fn(),
   useRBAC: jest.fn(),
+  useQueryParams: jest.fn(() => [{}]),
```

Comment:
You don't need to mock this hook and the Search component you just need to change your App :

```
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

jest.mock('@strapi/helper-plugin', () => ({
  ...jest.requireActual('@strapi/helper-plugin'),
  useTracking: jest.fn(() => ({ trackUsage: jest.fn() })),
  useNotification: jest.fn(),
  useRBAC: jest.fn(),
  CheckPermissions: ({ children }) => children,
}));


const makeApp = history => (
  <Router history={history}>
    <ThemeProvider theme={lightTheme}>
      <QueryClientProvider client={client}>
        <IntlProvider locale="en" messages={{ en: {} }} textComponent="span">
          <RoleListPage />
        </IntlProvider>
      </QueryClientProvider>
    </ThemeProvider>
  </Router>
);

```

And in your tests:

```js
const history = createMemoryHistory();
    const App = makeApp(history);
    const {
      container: { firstChild },
    } = render(App);
```